### ISSUE TITLE
dependency missing, scipy 1.7.1 package not available via pip

### ISSUE THREAD

**Issue by Last-Tiger-Eyes** [ksjdragon/squish#1]
**OS**:  Windows 10 Home, Version 21H1 (OS Build 19043.1586)
**Python Version**:  3.10.0
**Pip Version**:  22.0.4

```
C:\Users\redacted>pip install squish
Collecting squish
  Using cached squish-0.1.6.post2.tar.gz (361 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Discarding _https://files.pythonhosted.org/packages/93/51/0545cb9ac1feaa44e57189410376a06332a0a77e640d24ff697b91baaac8/squish-0.1.6.post2.tar.gz#sha256=de67f8da667437f49617efa2f338d5fb673eddd930c5d44be3546a4017b9ca33 (from https://pypi.org/simple/squish/) (requires-python:>=3.8)_: **Requested squish from https://files.pythonhosted.org/packages/93/51/0545cb9ac1feaa44e57189410376a06332a0a77e640d24ff697b91baaac8/squish-0.1.6.post2.tar.gz#sha256=de67f8da667437f49617efa2f338d5fb673eddd930c5d44be3546a4017b9ca33 has inconsistent version: filename has '0.1.6.post2', but metadata has '0.1.6'**
  Using cached squish-0.1.5.post2.tar.gz (353 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting matplotlib==3.4.3
  Using cached matplotlib-3.4.3.tar.gz (37.9 MB)
  Preparing metadata (setup.py) ... done
Collecting squish
  Using cached squish-0.1.5.post1.tar.gz (353 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached squish-0.1.5.tar.gz (354 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached squish-0.1.4.tar.gz (211 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached squish-0.1.3.tar.gz (211 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached squish-0.1.2.tar.gz (211 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached squish-0.1.1.tar.gz (211 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached squish-0.1.0.tar.gz (210 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
**ERROR: Cannot install squish==0.1.0, squish==0.1.1, squish==0.1.2, squish==0.1.3, squish==0.1.4, squish==0.1.5, squish==0.1.5.post1 and squish==0.1.5.post2 because these package versions have conflicting dependencies.**

The conflict is caused by:
    squish 0.1.5.post2 depends on scipy==1.7.1
    squish 0.1.5.post1 depends on scipy==1.7.1
    squish 0.1.5 depends on scipy==1.7.1
    squish 0.1.4 depends on scipy==1.7.1
    squish 0.1.3 depends on scipy==1.7.1
    squish 0.1.2 depends on scipy==1.7.1
    squish 0.1.1 depends on scipy==1.7.1
    squish 0.1.0 depends on scipy==1.7.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

**ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts**
```

Same error when I download and try to install from source:

```
C:\Users\redacted>py -m pip install -e C:\Users\redacted\AppData\Local\Programs\Python\Python310\Lib\site-packages\squish-0.1.6
Obtaining file:///C:/Users/redacted/AppData/Local/Programs/Python/Python310/Lib/site-packages/squish-0.1.6
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy>=1.21.2 in c:\users\redacted\appdata\local\programs\python\python310\lib\site-packages (from squish==0.1.6) (1.22.3)
Collecting matplotlib==3.4.3
  Using cached matplotlib-3.4.3.tar.gz (37.9 MB)
  Preparing metadata (setup.py) ... done
ERROR: Could not find a version that satisfies the requirement scipy==1.7.1 (from squish) (from versions: 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.13.1, 0.13.2, 0.13.3, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.17.0, 0.17.1, 0.18.0, 0.18.1, 0.19.0, 0.19.1, 1.0.0b1, 1.0.0rc1, 1.0.0rc2, 1.0.0, 1.0.1, 1.1.0rc1, 1.1.0, 1.2.0rc1, 1.2.0rc2, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.3.0rc1, 1.3.0rc2, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0rc1, 1.4.0rc2, 1.4.0, 1.4.1, 1.5.0rc1, 1.5.0rc2, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0rc1, 1.6.0rc2, 1.6.0, 1.6.1, 1.7.2, 1.7.3, 1.8.0rc1, 1.8.0rc2, 1.8.0rc3, 1.8.0rc4, 1.8.0)
ERROR: No matching distribution found for scipy==1.7.1
```

So I tried to install the missing dependency myself:

```
C:\Users\redacted>pip install scipy==1.7.1
ERROR: Could not find a version that satisfies the requirement scipy==1.7.1 (from versions: 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.13.1, 0.13.2, 0.13.3, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.17.0, 0.17.1, 0.18.0, 0.18.1, 0.19.0, 0.19.1, 1.0.0b1, 1.0.0rc1, 1.0.0rc2, 1.0.0, 1.0.1, 1.1.0rc1, 1.1.0, 1.2.0rc1, 1.2.0rc2, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.3.0rc1, 1.3.0rc2, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0rc1, 1.4.0rc2, 1.4.0, 1.4.1, 1.5.0rc1, 1.5.0rc2, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0rc1, 1.6.0rc2, 1.6.0, 1.6.1, 1.7.2, 1.7.3, 1.8.0rc1, 1.8.0rc2, 1.8.0rc3, 1.8.0rc4, 1.8.0)
ERROR: No matching distribution found for scipy==1.7.1
```

In the downloaded source files, I modified setup.py to have the lines:

```
[options]
zip_safe = False
packages = squish
python_requires = >= 3.8
install_requires = 
	numpy == 1.21.2
	scipy >= 1.7.1  # previously scipy==1.7.1
	matplotlib >= 3.4.3  # previously matplotlib==3.4.3
```

And now installation was successful.
I'm not sure if it's stable, though.
Can you advise if this workaround is appropriate?
(edited for formatting)

**Comment by Last-Tiger-Eyes** [comment]
Trying to install scipy 1.7.1 from source results in a failure (log attached for anyone interested), which is probably why it's no longer available via pip.
Can the scipy dependency be updated to an available/stable version?
[scipy_1.7.1_source_install_stacktrace.txt](https://github.com/ksjdragon/squish/files/8353751/scipy_1.7.1_source_install_stacktrace.txt)
(edited for formatting)

**Comment by ksjdragon** [comment]
Thanks for letting me know. That workaround should be fine. Until SciPy does a major change to its Voronoi computation library, nothing should break.

**Comment by ksjdragon** [comment]
Also, do you mind telling me what interest you have in this project?

**Comment by Last-Tiger-Eyes** [comment]
> Also, do you mind telling me what interest you have in this project?

I've worked with python based validation automation for a bit, and this project seemed like it might be a good option for automating some GUI applications. Was going to play around with it to see how useful it would be, but had trouble installing, so... here I am. ^_^

Thanks for getting back to me so quickly!

**Comment by ksjdragon** [comment]
Ah, that's a different library. I think the library you are looking for is `squishtest`. I used squish as a name because it wasn't taken on PyPi, never realized there was another large program already taking the name.  

Presumably this: https://kb.froglogic.com/squish/howto/using-squish-module-python-scripts-applications/. I don't believe they have a GitHub repo for their python package (or their program)

I found it pretty surprising that anyone would have looked up this project, since this is pretty small and the use case is as well.
I assume `pip install squishtest` will work.
